### ISSUE TITLE
assigment in do causes TypeError: Cannot read property 'value' of undefined

### ISSUE THREAD

**Issue by go-oleg** [za-creature/coffeescope#25]
### Code sample

``` coffeescript
myVar1 = 1
myVar2 = 2

do (myVar = myVar2) ->
  console.log(myVar)
```
### Output / stack trace

```
/Users/oseletsky/coffeescope/node_modules/coffeescope2/lib/ScopeLinter.js:155
          this.scope.identifierRead(arg.name.value, arg);
                                            ^

TypeError: Cannot read property 'value' of undefined
  at ScopeLinter.module.exports.ScopeLinter.visitCall (/Users/oseletsky/coffeescope/node_modules/coffeescope2/lib/ScopeLinter.js:155:45)
  at /Users/oseletsky/coffeescope/node_modules/coffeescope2/lib/ScopeLinter.js:5:59
  at ScopeLinter.module.exports.ScopeLinter.visit (/Users/oseletsky/coffeescope/node_modules/coffeescope2/lib/ScopeLinter.js:97:9)
  at /Users/oseletsky/coffeescope/node_modules/coffeescope2/lib/ScopeLinter.js:5:59
  at Block.exports.Base.Base.eachChild (/Users/oseletsky/coffeescope/node_modules/coffee-script/lib/coffee-script/nodes.js:199:17)
  at ScopeLinter.module.exports.ScopeLinter.visit (/Users/oseletsky/coffeescope/node_modules/coffeescope2/lib/ScopeLinter.js:99:14)
  at ScopeLinter.visit (/Users/oseletsky/coffeescope/node_modules/coffeescope2/lib/ScopeLinter.js:5:59)
  at /Users/oseletsky/coffeescope/node_modules/coffeescope2/lib/ScopeLinter.js:48:26
  at ScopeLinter.module.exports.ScopeLinter.newScope (/Users/oseletsky/coffeescope/node_modules/coffeescope2/lib/ScopeLinter.js:79:9)
  at ScopeLinter.newScope (/Users/oseletsky/coffeescope/node_modules/coffeescope2/lib/ScopeLinter.js:5:59)
  at ScopeLinter.module.exports.ScopeLinter.lint (/Users/oseletsky/coffeescope/node_modules/coffeescope2/lib/ScopeLinter.js:46:14)
  at ScopeLinter.lint (/Users/oseletsky/coffeescope/node_modules/coffeescope2/lib/ScopeLinter.js:5:59)
  at Coffeescope2.module.exports.Coffeescope2.lintAST (/Users/oseletsky/coffeescope/node_modules/coffeescope2/lib/index.js:34:38)
  at ASTLinter.module.exports.ASTLinter.lint (/Users/oseletsky/coffeescope/node_modules/coffeelint/lib/coffeelint.js:613:43)
  at Object.coffeelint.lint (/Users/oseletsky/coffeescope/node_modules/coffeelint/lib/coffeelint.js:370:67)
  at ErrorReport.module.exports.ErrorReport.lint (/Users/oseletsky/coffeescope/node_modules/coffeelint/lib/coffeelint.js:758:51)
  at lintFiles (/Users/oseletsky/coffeescope/node_modules/coffeelint/lib/commandline.js:98:19)
  at Object.<anonymous> (/Users/oseletsky/coffeescope/node_modules/coffeelint/lib/commandline.js:251:21)
  at Object.<anonymous> (/Users/oseletsky/coffeescope/node_modules/coffeelint/lib/commandline.js:256:4)
  at Module._compile (module.js:541:32)
  at Object.Module._extensions..js (module.js:550:10)
  at Module.load (module.js:458:32)
  at tryModuleLoad (module.js:417:12)
  at Function.Module._load (module.js:409:3)
  at Module.require (module.js:468:17)
  at require (internal/module.js:20:19)
  at Object.<anonymous> (/Users/oseletsky/coffeescope/node_modules/coffeelint/bin/coffeelint:34:5)
  at Module._compile (module.js:541:32)
  at Object.Module._extensions..js (module.js:550:10)
  at Module.load (module.js:458:32)
  at tryModuleLoad (module.js:417:12)
  at Function.Module._load (module.js:409:3)
  at Function.Module.runMain (module.js:575:10)
  at startup (node.js:160:18)
  at node.js:449:3
```
### coffeelint.json

``` json
{
  "check_scope": {
    "module": "coffeescope2",
    "level": "error",
    "undefined": true,
    "shadow": false
  }
}
```

The following works fine

```
myVar1 = 1
myVar2 = 2

do (myVar) ->
  console.log(myVar)
```

So the problem is in the assignment within the `do`.

Thanks and please let me know if you can't reproduce or want me to try something else.

**Comment by za-creature** [comment]
Hey, thanks for the report!

I actually ran into this myself a few days ago, but the past few days have been a bit too exciting for me to prep a release.

The fix for this has been released as 0.4.0 because ab7372f21df3b7239691d4a64a7dd6a7aba28e23 might break compat.

**Comment by go-oleg** [comment]
You are awesome. Thanks!
